### PR TITLE
DlgOneWayFrequencies

### PR DESCRIPTION
**Pull request by stevenndungu**

### CODE REVIEW

File: instat/dlgOneWayFrequencies.vb [hunk 1]
```diff
@@ -87,10 +94,13 @@ Public Class dlgOneWayFrequencies
         ucrReceiverOneWayFreq.SetParameterIsRFunction()
         clsSjtFreq.SetRCommand("sjPlot::sjt.frq")
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: So I recommend we do "sjPlot::sjtab" here and do sjPlot::sjplot where we currently have sjp.frq

File: instat/dlgOneWayFrequencies.vb [hunk 2]
```diff
@@ -51,6 +53,7 @@ Public Class dlgOneWayFrequencies
 
         ucrReceiverOneWayFreq.Selector = ucrSelectorOneWayFreq
         ucrReceiverOneWayFreq.SetMeAsReceiver()
+        ucrReceiverOneWayFreq.SetDataType("numeric")
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: I thought we were accepting all variables?

**Comment by stevenndungu** [comment]
@africanmathsinitiative/developers this is ready for viewing and reviewing as it awaits more changes

File: instat/dlgOneWayFrequencies.vb [hunk 3]
```diff
@@ -54,49 +59,54 @@ Public Class dlgOneWayFrequencies
         ucrReceiverOneWayFreq.SetParameter(New RParameter("data", 1))
         ucrReceiverOneWayFreq.SetParameterIsRFunction()
 
+        ucrReceiverWeights.Selector = ucrSelectorOneWayFreq
+        ucrReceiverWeights.SetParameter(New RParameter("weight.by", 3))
+        ucrReceiverWeights.SetParameterIsRFunction()
+        ucrReceiverWeights.SetDataType("numeric")
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: This adds the parameter "weight.by", however, when I check the checkbox this parameter is not added. I thought the checkbox and "weight.by" option was meant to bring up another receiver which had this parameter?

File: instat/dlgOneWayFrequencies.vb [hunk 4]
```diff
@@ -54,49 +59,54 @@ Public Class dlgOneWayFrequencies
...
-        ' ucrChkGraph.SetParameter(New RParameter("type"))
+        ucrChkTable.AddFunctionNamesCondition(True, "sjPlot::sjt.frq")
 
+        ucrChkWeights.SetText("Weights")
+        ucrChkWeights.SetParameter(ucrReceiverWeights.GetParameter(), bNewChangeParameterValue:=False, bNewAddRemoveParameter:=True)
+        ucrChkWeights.AddToLinkedControls(ucrReceiverWeights, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+
+        ucrChkGraph.SetText("Graph")
         ucrChkFlip.SetText("Flip Coordinates")
         ucrChkFlip.SetParameter(New RParameter("coord.flip"), bNewChangeParameterValue:=True, bNewAddRemoveParameter:=True, strNewValueIfChecked:="TRUE", strNewValueIfUnchecked:="FALSE")
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: This parameter wasn't running for me. Additionally, should this option only been enabled if graph or table is checked?

File: instat/dlgOneWayFrequencies.vb [hunk 5]
```diff
@@ -54,49 +59,54 @@ Public Class dlgOneWayFrequencies
...
         ucrChkFlip.SetRDefault("FALSE")
 
-        ucrSaveGraph.SetPrefix("one_way_freq")
-        ucrSaveGraph.SetSaveTypeAsGraph()
-        ucrSaveGraph.SetDataFrameSelector(ucrSelectorOneWayFreq.ucrAvailableDataFrames)
-        ucrSaveGraph.SetCheckBoxText("Save Graph")
-        ucrSaveGraph.SetIsComboBox()
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: Why did you remove the ucrSave code?

File: instat/dlgOneWayFrequencies.vb [hunk 6]
```diff
@@ -108,7 +118,7 @@ Public Class dlgOneWayFrequencies
             ucrReceiverOneWayFreq.SetParameter(New RParameter("data", 1))
             ucrReceiverOneWayFreq.SetParameterIsRFunction()
             ucrBase.clsRsyntax.SetBaseRFunction(clsSjtFreq)
-        Else
+        ElseIf ucrChkGraph.Checked Then
             ucrReceiverOneWayFreq.SetParameter(New RParameter("var.cnt", 1))
             ucrReceiverOneWayFreq.SetParameterIsRFunction()
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: This line is being run in both the "If" and in the "Else"

File: instat/dlgOneWayFrequencies.vb (hunk 5 above)
Comment (stevenndungu) [inline instat/dlgOneWayFrequencies.vb]: Well, will add them.. though currently do not see its function

**Comment by stevenndungu** [comment]
Suggested changes made this ready for re-review.

**Comment by Lunalo** [comment]
TestOkEnabled need to be edited .....Some controls are missing in valueChanged.
Allignment of controls on left need to be 10 by x
Save graph not resetting

File: instat/dlgOneWayFrequencies.vb [hunk 7]
```diff
@@ -153,7 +153,7 @@ Public Class dlgOneWayFrequencies
         ChangeBaseFunction()
     End Sub
 
-    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneWayFreq.ControlContentsChanged
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneWayFreq.ControlContentsChanged, ucrChkWeights.ControlContentsChanged, ucrReceiverWeights.ControlContentsChanged, ucrSaveGraph.ControlContentsChanged
```
Comment (Lunalo) [inline instat/dlgOneWayFrequencies.vb]: You are missing event for ucrChkWeights

File: instat/dlgOneWayFrequencies.vb [hunk 8]
```diff
@@ -108,7 +108,7 @@ Public Class dlgOneWayFrequencies
     End Sub
 
     Private Sub TestOkEnabled()
-        If Not ucrReceiverOneWayFreq.IsEmpty() Then
+        If Not ucrReceiverOneWayFreq.IsEmpty() AndAlso ucrSaveGraph.IsComplete Then
```
Comment (Lunalo) [inline instat/dlgOneWayFrequencies.vb]: ` If Not ucrReceiverOneWayFreq.IsEmpty() Then
            If Not ucrChkWeights.Checked Then
                ucrBase.OKEnabled(True)
            Else
                If Not ucrReceiverWeights.IsEmpty Then
                    ucrBase.OKEnabled(True)
                Else
                    ucrBase.OKEnabled(False)
                End If
            End If
        Else
            ucrBase.OKEnabled(False)
        End If`

It needs to be adjusted

**Comment by Lunalo** [comment]
@dannyparsons 

You can merge this so that all the changes are made together

File: instat/dlgOneWayFrequencies.vb [hunk 9]
```diff
@@ -37,78 +37,90 @@ Public Class dlgOneWayFrequencies
...
-        ucrSaveGraph.SetIsComboBox()
     End Sub
 
     Private Sub SetDefaults()
         clsSjtFreq = New RFunction
         clsSjpFrq = New RFunction
         ucrSelectorOneWayFreq.Reset()
-        ucrSaveGraph.Reset()
         ucrReceiverOneWayFreq.SetParameter(New RParameter("data", 1))
         ucrReceiverOneWayFreq.SetParameterIsRFunction()
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: This is in the initialise. How come you have repeated these two lines here?

File: instat/dlgOneWayFrequencies.vb [hunk 10]
```diff
@@ -37,78 +37,90 @@ Public Class dlgOneWayFrequencies
...
     Private Sub ChangeBaseFunction()
-        If ucrChkTable.Checked Then
+        If rdoTable.Checked Then
             ucrReceiverOneWayFreq.SetParameter(New RParameter("data", 1))
             ucrReceiverOneWayFreq.SetParameterIsRFunction()
             ucrBase.clsRsyntax.SetBaseRFunction(clsSjtFreq)
-        Else
+        ElseIf rdoGraph.Checked Then
             ucrReceiverOneWayFreq.SetParameter(New RParameter("var.cnt", 1))
             ucrReceiverOneWayFreq.SetParameterIsRFunction()
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: Since in both parts of the IF statement you have set `ucrReceiverOneWayFreq.SetParameterIsRFunction()` then you can do this once after the End If

File: instat/dlgOneWayFrequencies.vb [hunk 11]
```diff
@@ -37,78 +37,90 @@ Public Class dlgOneWayFrequencies
...
-        ' ucrChkGraph.SetParameter(New RParameter("type"))
+        ucrPnlFreqDisplay.AddFunctionNamesCondition(rdoTable, "sjPlot::sjt.frq")
+        ucrPnlFreqDisplay.AddFunctionNamesCondition(rdoGraph, "sjPlot::sjp.frq")
+
+        ucrChkWeights.SetText("Weights")
+        ucrChkWeights.SetParameter(ucrReceiverWeights.GetParameter(), bNewChangeParameterValue:=False, bNewAddRemoveParameter:=True)
+        ucrChkWeights.AddToLinkedControls(ucrReceiverWeights, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+
+        ucrPnlFreqDisplay.AddToLinkedControls(ucrPnlSort, {rdoTable}, bNewLinkedDisabledIfParameterMissing:=True, bNewLinkedAddRemoveParameter:=True)
+        ucrPnlFreqDisplay.AddToLinkedControls(ucrChkFlip, {rdoGraph}, bNewLinkedDisabledIfParameterMissing:=True, bNewLinkedAddRemoveParameter:=True)
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: I thought the options for None/Asc/Desc applied to both rdoTable and rdoGraph? It was working for me earlier I think? Why did you disable this

File: instat/dlgOneWayFrequencies.vb [hunk 12]
```diff
@@ -122,14 +134,19 @@ Public Class dlgOneWayFrequencies
...
+            ucrReceiverOneWayFreq.SetMeAsReceiver()
         End If
+    End Sub
+
+    Private Sub AllControls_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrPnlFreqDisplay.ControlValueChanged
         ChangeBaseFunction()
     End Sub
 
-    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneWayFreq.ControlContentsChanged, ucrSaveGraph.ControlContentsChanged
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneWayFreq.ControlContentsChanged, ucrReceiverWeights.ControlContentsChanged
```
Comment (lilyclements) [inline instat/dlgOneWayFrequencies.vb]: You need the checkbox in here too
